src/util.ts

```diff
@@ -393,3 +393,27 @@ const addGraphTotals = (
   )
   return { totalTxs, totalUsd }
 }
+
+export const movingAveDataSort = (
+  data: Data[]
+): Array<{ date: string; allTxs: number }> => {
+  const aveArray: Array<{ date: string; allTxs: number }> = []
+  if (Array.isArray(data) === false) return aveArray
+  const dataLength = data.length
+  let sevenDaySum: number = 0
+  for (let i = 0; i < dataLength; i++) {
+    for (let j = i; j >= i - 6; j--) {
+      if (j < 0) {
+        sevenDaySum += 0
+        continue
+      }
+      const currentData: Data = data[j]
+      sevenDaySum += currentData.allTxs
+    }
+    if (typeof sevenDaySum !== 'number' || isNaN(sevenDaySum)) return []
+    const sevenDayAve: number = Math.round(sevenDaySum / 7)
+    aveArray.push({ date: data[i].date, allTxs: sevenDayAve })
+    sevenDaySum = 0
```

Comment:
This line can be deleted if line 403 is moved inside the `for` loop on line 404. See comment above.